crates/ide/src/inlay_hints/closing_brace.rs

```diff
@@ -154,11 +189,124 @@ pub(super) fn hints(
     None
 }
 
+fn fn_qualifiers(func: &ast::Fn) -> String {
+    let mut qualifiers = String::new();
+    if func.const_token().is_some() {
+        qualifiers.push_str("const ");
+    }
+    if func.async_token().is_some() {
+        qualifiers.push_str("async ");
+    }
+    if func.unsafe_token().is_some() {
+        qualifiers.push_str("unsafe ");
+    }
+    qualifiers
+}
+
+fn keyword_with_condition(
+    keyword: &str,
+    condition: Option<ast::Expr>,
+    config: &InlayHintsConfig<'_>,
+) -> String {
+    if let Some(expr) = condition {
+        return format!("{keyword} {}", snippet_from_node(expr.syntax(), config));
+    }
+    keyword.to_owned()
+}
+
+fn format_for_label(for_expr: &ast::ForExpr, config: &InlayHintsConfig<'_>) -> Option<String> {
+    let pat = for_expr.pat()?;
+    let iterable = for_expr.iterable()?;
+    Some(format!(
+        "for {} in {}",
+        snippet_from_node(pat.syntax(), config),
+        snippet_from_node(iterable.syntax(), config)
+    ))
+}
+
+fn format_match_label(
+    match_expr: &ast::MatchExpr,
+    config: &InlayHintsConfig<'_>,
+) -> Option<String> {
+    let expr = match_expr.expr()?;
+    Some(format!("match {}", snippet_from_node(expr.syntax(), config)))
+}
+
+fn label_for_if_block(
+    if_expr: &ast::IfExpr,
+    block: &ast::BlockExpr,
+    config: &InlayHintsConfig<'_>,
+) -> Option<String> {
+    if if_expr.then_branch().is_some_and(|then_branch| then_branch.syntax() == block.syntax()) {
+        Some(keyword_with_condition("if", if_expr.condition(), config))
+    } else if matches!(
+        if_expr.else_branch(),
+        Some(ast::ElseBranch::Block(else_block)) if else_block.syntax() == block.syntax()
+    ) {
+        Some("else".into())
+    } else {
+        None
+    }
+}
+
+fn format_let_else_label(let_else: &ast::LetElse, config: &InlayHintsConfig<'_>) -> Option<String> {
+    let stmt = let_else.syntax().parent().and_then(ast::LetStmt::cast)?;
+    let pat = stmt.pat()?;
+    let initializer = stmt.initializer()?;
+    Some(format!(
+        "let {} = {} else",
+        snippet_from_node(pat.syntax(), config),
+        snippet_from_node(initializer.syntax(), config)
+    ))
+}
+
+fn snippet_from_node(node: &SyntaxNode, config: &InlayHintsConfig<'_>) -> String {
+    let mut text = node.text().to_string();
+    if text.contains('\n') {
+        return ELLIPSIS.into();
+    }
+
+    let Some(limit) = config.max_length else {
+        return text;
+    };
+    if limit == 0 {
+        return ELLIPSIS.into();
+    }
+
+    if text.len() <= limit {
+        return text;
+    }
+
+    let mut boundary = limit.min(text.len());
+    while boundary > 0 && !text.is_char_boundary(boundary) {
```

Comment:
Why not `floor_char_boundary()`, as I suggested?

Reply:
Just found that I haven't updated the rust version... fixed.